cmd/snap-failure/cmd_snapd.go

```diff
@@ -185,10 +185,7 @@ func (c *cmdSnapd) Execute(args []string) error {
 		}
 	}
 	// start previous snapd
-	// TODO:FDEM:FIX: we should either run through systemd with
-	// KeyringMode=shared or make the re-seeding not initialize
-	// fdestate
-	cmd := runCmd(snapdPath, nil, []string{"SNAPD_REVERT_TO_REV=" + prevRev, "SNAPD_DEBUG=1"})
+	cmd := runCmd("systemd-run", []string{"--collect", "--wait", "--property=KeyringMode=shared", fmt.Sprintf("--setenv=SNAPD_REVERT_TO_REV=%s", prevRev), "--setenv=SNAPD_DEBUG=1", "--", snapdPath}, []string{})
```

Comment:
--wait is from systemd 232+ and I think it should work with core18, but please double check --collect
```suggestion
	cmd := runCmd("systemd-run", 
		[]string{"--collect", "--wait", 
				"--property=KeyringMode=shared",
				"--setenv=SNAPD_REVERT_TO_REV="+prevRev,
				 "--setenv=SNAPD_DEBUG=1", 
				 "--", snapdPath,
		}, []string{})
```

Reply:
Bionic should have 237. And this should have been added in 236.